utils/streams.js

```diff
@@ -0,0 +1,113 @@
+const fs = require('fs');
+const https = require('https');
+const program = require('commander');
+const through = require('through2');
+const csvjson = require('csvjson');
+
+
+
+program.option('-a, --action <string>', 'read action')
+.option('-f, --file <string>', 'read file')
+.option('-p, --path <string>', 'path for directory or file')
+.parse(process.argv);
+
+/* Validating arguments and mapping actions */
+
+if (process.argv.length < 3) {
+    program.help();
+} else if (process.argv[2] == '-h' || process.argv[2] == '--help') {
+    program.help();       
+} else{
+    switch(program.action) {
+        case 'reverse': reverse();
+            break;
+        case 'transform': transform();
+            break;
+        case 'outputFile': outputFile();
```

Comment:
Don't you need to pass a `program.file` or `program.path`  as an argument to the `outputFile()` and other following functions?

Reply:
Also to use switch statement is not the best practice. The better approach is just define an object with properties that are defined functions and access these functions via bracket notation:

`const functions = { reverse, transform, outputFile, convertFromFile };`

`functions[program.actions]()`

Reply:
Resolved